### PR TITLE
expression: Fix greatest and least function lost decimal precision compared with MySQL

### LINKED ISSUE #25765: Greatest and least function lost decimal precision compared with MySQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Master branch

`select greatest(1.0, 2);`

### 2. What did you expect to see? (Required)
MySQL

```
mysql> select greatest(1.0, 2);
+------------------+
| greatest(1.0, 2) |
+------------------+
|              2.0 |
+------------------+
1 row in set (0.00 sec)

```

### 3. What did you see instead (Required)
TiDB 

```
mysql> select greatest(1.0, 2);
+------------------+
| greatest(1.0, 2) |
+------------------+
|                2 |
+------------------+
1 row in set (0.00 sec)

```

### 4. What is your TiDB version? (Required)

Master branch

### PR DESCRIPTION
**Pull request by yuqi1129**
### What problem does this PR solve?

Issue Number: close #25765 

Problem Summary:
Greatest and least function lost decimal precision compared with MySQL
see:
`select greatest(1.0, 2)` return `2` while MySQL returns `2.0`

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- `No release note`

### CODE REVIEW

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -45,14 +45,32 @@ func (b *builtinGreatestDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *c
...
+			d64s[i].SetDigitsFrac(maxDigitFrac)
+			d64s[i].SetDigitsInt(maxDigitInt)
+			d64s[i].SetResultFrac(maxResultFrac)
 		}
 	}
 	return nil
 }
 
+// maxInt8 return max value of two int8 value
+func maxInt8(x int8, y int8) int8 {
```
Comment (lzmhhh123) [inline expression/builtin_compare_vec.go:66]: remove it and use `mathutil.MaxInt8`

Comment (yuqi1129) [reply expression/builtin_compare_vec.go:66]: OK

File: expression/builtin_compare_vec.go [hunk 2]
```diff
@@ -16,6 +16,8 @@ package expression
 import (
 	"strings"
 
+	"github.com/cznic/mathutil"
+
```
Comment (lzmhhh123) [inline expression/builtin_compare_vec.go:20]: Remove the line and resort the import.

**Review by lzmhhh123 — APPROVED** [review]

**Comment by yuqi1129** [comment]
@ichn-hu , Can you help to review this PR ?

**Review by ichn-hu — APPROVED** [review]

**Pull request merged**